src/ser.rs

```diff
@@ -401,115 +209,113 @@ where
     }
 
     #[inline]
-    fn serialize_newtype_variant<T>(
+    fn serialize_newtype_variant<T: Serialize + ?Sized>(
         self,
-        name: &'static str,
-        variant_index: u32,
+        _name: &'static str,
+        _variant_index: u32,
         variant: &'static str,
         value: &T,
-    ) -> Result<()>
-    where
-        T: ?Sized + ser::Serialize,
-    {
-        if self.enum_as_map {
-            self.write_u64(5, 1u64)?;
-            variant.serialize(&mut *self)?;
-        } else {
-            self.writer.write_all(&[4 << 5 | 2]).map_err(|e| e.into())?;
-            self.serialize_unit_variant(name, variant_index, variant)?;
-        }
+    ) -> Result<Self::Ok, Self::Error> {
+        enc::MapStartBounded(1).encode(&mut self.writer)?;
+        variant.encode(&mut self.writer)?;
         value.serialize(self)
     }
 
     #[inline]
-    fn serialize_seq(self, len: Option<usize>) -> Result<CollectionSerializer<'a, W>> {
-        self.serialize_collection(4, len)
+    fn serialize_seq(self, len: Option<usize>) -> Result<Self::SerializeSeq, Self::Error> {
+        if let Some(len) = len {
+            enc::ArrayStartBounded(len).encode(&mut self.writer)?;
+        } else {
+            enc::ArrayStartUnbounded.encode(&mut self.writer)?;
+        }
+        Ok(Collect {
+            bounded: len.is_some(),
+            ser: self,
+        })
     }
 
     #[inline]
-    fn serialize_tuple(self, len: usize) -> Result<&'a mut Serializer<W>> {
-        self.write_u64(4, len as u64)?;
-        Ok(self)
+    fn serialize_tuple(self, len: usize) -> Result<Self::SerializeTuple, Self::Error> {
+        enc::ArrayStartBounded(len).encode(&mut self.writer)?;
+        Ok(BoundedCollect { ser: self })
     }
 
     #[inline]
     fn serialize_tuple_struct(
         self,
         _name: &'static str,
         len: usize,
-    ) -> Result<&'a mut Serializer<W>> {
+    ) -> Result<Self::SerializeTupleStruct, Self::Error> {
         self.serialize_tuple(len)
     }
 
     #[inline]
     fn serialize_tuple_variant(
         self,
-        name: &'static str,
-        variant_index: u32,
+        _name: &'static str,
```

Comment:
why are these ignored now?

Reply:
`serde_cbor` had an option for a "packed" representation, where they weren't represented as maps. We never used that packed representation.